drivers/usb/device/usb_dc_smartbond.c

```diff
@@ -0,0 +1,1398 @@
+			REG_SET_VAL(USB_TXC0_REG, USB_TOGGLE_TX0,
+				    ep0_in_state->data1);
+			if (ep0_in_state->transferred == ep0_in_state->total_len) {
+				/* For control endpoint get ready for ACK stage
+				 * from host.
+				 */
+				ep0_out_state = usb_dc_get_ep_out_state(EP0_IDX);
+				ep0_out_state->transferred = 0;
+				ep0_out_state->total_len = 0;
+				ep0_out_state->last_packet_size = 0;
+				REG_SET_BIT(USB_RXC0_REG, USB_RX_EN);
+
+				atomic_clear(&ep0_in_state->busy);
+				ep0_in_state->cb(EP0_IN, USB_DC_EP_DATA_IN);
+				return;
+			}
+		} else {
+			/* Start from the beginning */
+			ep0_in_state->last_packet_size = 0;
+		}
+		start_tx_packet(ep0_in_state);
+	}
+}
+
+static void handle_epx_rx_ev(uint8_t ep_idx)
+{
+	uint32_t rxs;
+	int fifo_bytes;
+	struct smartbond_ep_state *ep_state = usb_dc_get_ep_out_state(ep_idx);
+	struct smartbond_ep_reg_set *regs = ep_state->regs;
+
+	do {
+		rxs = regs->rxs;
+
+		if (GET_BIT(rxs, USB_USB_RXS1_REG_USB_RX_ERR)) {
+			regs->rxc |= USB_USB_RXC1_REG_USB_FLUSH_Msk;
+			ep_state->last_packet_size = 0;
+			if (dev_state.dma_ep[0] == ep_state) {
+				/* Stop DMA */
+				RX_DMA_REGS->DMAx_CTRL_REG &= ~DMA_DMA0_CTRL_REG_DMA_ON_Msk;
+				/* Restart DMA since packet was dropped,
+				 * all parameters should still work.
+				 */
+				RX_DMA_REGS->DMAx_CTRL_REG |= DMA_DMA0_CTRL_REG_DMA_ON_Msk;
+			}
+			break;
+		}
+		if (dev_state.dma_ep[0] == ep_state) {
+			/* Disable DMA and update last_packet_size
+			 * with what DMA reported.
+			 */
+			RX_DMA_REGS->DMAx_CTRL_REG &= ~DMA_DMA0_CTRL_REG_DMA_ON_Msk;
+			ep_state->last_packet_size = RX_DMA_REGS->DMAx_IDX_REG;
+			/*
+			 * When DMA did not finished (packet was smaller then MPS),
+			 * DMAx_IDX_REG holds exact number of bytes transmitted. When DMA
+			 * finished value in DMAx_IDX_REG is one less then actual number of
+			 * transmitted bytes.
+			 */
+			if (ep_state->last_packet_size == RX_DMA_REGS->DMAx_LEN_REG) {
+				ep_state->last_packet_size++;
+			}
+			/* Release DMA to use by other endpoints. */
+			dev_state.dma_ep[0] = NULL;
+		}
+		fifo_bytes = GET_BIT(rxs, USB_USB_RXS1_REG_USB_RXCOUNT);
+		/*
+		 * FIFO maybe empty if DMA read it before or
+		 * it's final iteration and function already read all
+		 * that was to read.
+		 */
+		if (fifo_bytes > 0) {
+			fifo_bytes = read_rx_fifo(ep_state, fifo_bytes);
+		}
+		if (GET_BIT(rxs, USB_USB_RXS1_REG_USB_RX_LAST)) {
+			if (!ep_state->iso &&
+			    GET_BIT(rxs, USB_USB_RXS1_REG_USB_TOGGLE_RX) !=
+			    ep_state->data1) {
+				/* Toggle bit does not match discard packet */
+				regs->rxc |= USB_USB_RXC1_REG_USB_FLUSH_Msk;
+				ep_state->last_packet_size = 0;
+				/* Re-enable reception */
+				start_rx_packet(ep_state);
+			} else {
+				ep_state->data1 ^= 1;
+				atomic_clear(&ep_state->busy);
+				ep_state->cb(ep_state->ep_addr,
+					     USB_DC_EP_DATA_OUT);
+			}
+		}
+	} while (fifo_bytes > FIFO_READ_THRESHOLD);
+}
+
+static void handle_rx_ev(void)
+{
+	if (USB->USB_RXEV_REG & 1) {
```

Comment:
Zephyr has `BIT()` macro for specifying bits. Even better would be to have some definitions for register bits.

Reply:
Hi @tmon-nordic 

I updated this function, handle_rx_ev and handle_fifo_warning to use BIT()